src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -272,6 +321,53 @@ private static void ScanForExpiredItems(MemoryCache cache)
             }
         }
 
+        private bool UpdateCacheSizeExceedsCapacity(CacheEntry entry, out bool cacheSizeUpdated)
```

Comment:
Can you eliminate cacheSizeUpdated? e.g. if it exceed capacity, immediately subtract it again. Then the only info you need out of this method is the return bool.